app/controllers/collections_controller.rb

```diff
@@ -338,4 +351,80 @@ def write_item_metadata(corpus_dir, item_json)
     {:identifier => item_json["identifier"], :rdf_file => rdf_file}
   end
 
+  # Deletes statements with the item's URI from Sesame
+  def delete_item_from_sesame(item, repository)
+    item_subject = RDF::URI.new(item.uri)
+    item_query = RDF::Query.new do
+      pattern [item_subject, :predicate, :object]
+    end
+    item_statements = repository.query(item_query)
+    item_statements.each do |item_statement|
+      repository.delete(RDF::Statement(item_subject, item_statement[:predicate], item_statement[:object]))
+    end
+  end
+
+  # Deletes statements with the document's derived URI from Sesame
+  def delete_document_from_sesame(document, repository)
+    document_query = RDF::Query.new do
+      pattern [:subject, MetadataHelper::SOURCE, RDF::URI.new("file://#{document.file_path}")]
+      pattern [:subject, MetadataHelper::IDENTIFIER, "#{document.file_name}"]
+    end
+    document_URIs = repository.query(document_query)
+    if document_URIs.count == 1
+      document_URI = document_URIs.first[:subject]
+      document_statements_query = RDF::Query.new do
+        pattern [document_URI, :predicate, :object]
+      end
+      document_statements = repository.query(document_statements_query)
+      document_statements.each do |document_statement|
+        repository.delete(RDF::Statement(document_URI, document_statement[:predicate], document_statement[:object]))
+      end
+    else
+      Rails.logger.error "Cannot delete document RDF as multiple distinct document URIs match the document file name and path"
+      document_URIs.each do |statement|
+        Rails.logger.debug "#{RDF::Statement(statement[:subject], MetadataHelper::SOURCE, RDF::URI.new("file://#{document.file_path}"))}"
+        Rails.logger.debug "#{RDF::Statement(statement[:subject], MetadataHelper::IDENTIFIER, "#{document.file_name}")}"
+      end
+    end
+  end
+
+  # Removes an item and its documents from the database, filesystem, Sesame and Solr
+  def remove_item(item, collection, corpus_dir)
+    # Remove from filesystem
```

Comment:
I'd recommend splitting this into 3 methods:
- `delete_from_filesystem`
- `delete_from_sesame`
- `delete_from_solr`